tests/testAstrometrySourceSelector.py

```diff
@@ -0,0 +1,121 @@
+
+
+class TestAstrometrySourceSelector(lsst.utils.tests.TestCase):
+    def setUp(self):
+        schema = TestDataset.makeMinimalSchema()
+        schema.addField("truth_flag", type="Flag")  # for slot_Centroid_flag
+        schema.addField("slot_ApFlux_flux", type=float)
+        schema.addField("slot_ApFlux_fluxSigma", type=float)
+        for flag in badFlags:
+            schema.addField(flag, type="Flag")
+
+        self.src = table.SourceCatalog(schema)
+        self.sourceSelector = sourceSelector.sourceSelectorRegistry['astrometry']()
+
+    def tearDown(self):
+        del self.src
+        del self.sourceSelector
+
+    def testSelectSources_good(self):
+        for i in range(5):
+            add_good_source(self.src, i)
+        result = self.sourceSelector.selectSources(self.src)
+        # TODO: assertEqual doesn't work right on source catalogs.
+        # self.assertEqual(result.sourceCat, self.src)
+        for x in self.src['id']:
+            self.assertIn(x, result.sourceCat['id'])
+
+    def testSelectSources_bad(self):
+        for i, flag in enumerate(badFlags):
+            add_good_source(self.src, i)
+            self.src[i].set(flag, True)
+        result = self.sourceSelector.selectSources(self.src)
+        for i, x in enumerate(self.src['id']):
+            self.assertNotIn(x, result.sourceCat['id'], "should not have found %s"%badFlags[i])
+
+    def testSelectSources_bad_centroid(self):
+        add_good_source(self.src, 1)
+        self.src[0].set('slot_Centroid_x', np.nan)
+        result = self.sourceSelector.selectSources(self.src)
+        self.assertNotIn(self.src['id'][0], result.sourceCat['id'])
+
+    def testSelectSources_is_parent(self):
+        add_good_source(self.src, 1)
+        self.src[0].set('parent', 1)
+        result = self.sourceSelector.selectSources(self.src)
+        self.assertNotIn(self.src['id'][0], result.sourceCat['id'])
+
+    def testSelectSources_has_children(self):
+        add_good_source(self.src, 1)
+        self.src[0].set('deblend_nChild', 1)
+        result = self.sourceSelector.selectSources(self.src)
+        self.assertNotIn(self.src['id'][0], result.sourceCat['id'])
+
+    def testSelectSources_highSN_cut(self):
+        add_good_source(self.src, 1)
+        add_good_source(self.src, 2)
+        self.src['slot_ApFlux_flux'][0] = 20.
+        self.src['slot_ApFlux_flux'][1] = 1000.
+
+        self.sourceSelector.config.minSnr = 100
+        result = self.sourceSelector.selectSources(self.src)
+        self.assertNotIn(self.src[0]['id'], result.sourceCat['id'])
+        self.assertIn(self.src[1]['id'], result.sourceCat['id'])
+
+    def testSelectSources_no_SN_cut(self):
+        self.sourceSelector.config.minSnr = 0
+        add_good_source(self.src, 1)
+        self.src['slot_ApFlux_flux'][0] = 0
+        result = self.sourceSelector.selectSources(self.src)
+        self.assertIn(self.src[0]['id'], result.sourceCat['id'])
+
+
+# for MemoryTestCase
```

Comment:
This is the new test boilerplate, described in https://sqr-012.lsst.io/. TestSuite is dead, long live lsst.utils.tests.init()!